packages/web-components/src/search/components/product-search-results.ts

```diff
@@ -0,0 +1,117 @@
+        window.addEventListener(Events.dimPreviousResult, () => this.handleDimPreviousResultEvent());
+        window.addEventListener(Events.searchingForProductsCompleted, () => this.handleSearchingForProductsCompletedEvent());
+    }
+
+    disconnectedCallback(): void {
+        window.removeEventListener(Events.showLoadingSpinner, this.handleShowLoadingSpinnerEvent);
+        window.removeEventListener(Events.dimPreviousResult, this.handleDimPreviousResultEvent);
+        window.removeEventListener(Events.searchingForProductsCompleted, this.handleSearchingForProductsCompletedEvent);
+
+        super.disconnectedCallback();
+    }
+
+    handleShowLoadingSpinnerEvent() {
+        this.showLoadingSpinner = true;
+    }
+
+    handleDimPreviousResultEvent() {
+        this.showDimmingOverlay = true;
+    }
+
+    handleSearchingForProductsCompletedEvent() {
+        this.showLoadingSpinner = false;
+        this.showDimmingOverlay = false;
+    }
+
+    render() {
+        const localization = getRelewiseUISearchOptions()?.localization?.searchResults;
+        return html`
+            <div class="rw-result-container">
+                ${this.showDimmingOverlay ? html`<div class="rw-dimming-overlay"></div>`: nothing}
+                ${this.products.length > 0 ? html`
+                    <div class="rw-product-grid">
+                        ${this.products.map(product => {
+                            return html`<relewise-product-tile class="rw-product-tile" .product=${product}></relewise-product-tile>`;
+                        })
+                        }
+                    </div>
+                ` : html`
+                        ${!this.showDimmingOverlay && !this.showLoadingSpinner ? html`
+                            <span class="rw-no-results-message">${localization?.noResults ?? 'No results found'}</span>
+                        ` : nothing}
+                `}
+                ${this.showLoadingSpinner ? html`
+                    <div class="rw-loading-spinner-container"><relewise-loading-spinner></relewise-loading-spinner></div>
+                ` : nothing}
+            </div>
+        `;
+    }
+
+    static styles = [theme, css`
+        :host {
+            font-family: var(--font);
+        }
+
+        .rw-result-container {
+            position: relative;
+        }
+
+        .rw-loading-spinner-container {
+            display: flex;
+            justify-content: center;
+            margin: 1rem;
+        }
+
+        .rw-product-grid {
+            display: grid;
+            grid-template-columns: repeat(2,1fr);
+            gap: 1rem;
+        }
+
+        .rw-product-tile {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+        }
+
+        .rw-no-results-message {
+            position: fixed;
+            left: 50%;
+            transform: translate(-50%, -50%);
+            text-align: center;
```

Comment:
Kan du ikke bruge flex i stedet for position fixed til dette?

Reply:
Check 👍